### ISSUE TITLE
calc.metrics

### ISSUE THREAD

**Issue by bcow** [PecanProject/pecan#610]
This function does the actual calculation of the benchmarking scores and can be done locally or on a remote server. 

input
- input.data.path
- formats.table
- variables (optinal)
- dates (optional)
- sites (optional)
- output directory that will have all sub directories for each run
will ultimately need to find these folders out of all content in the directory
- metrics

output:
- data.frame (or list object) of metric scores 

operation:
- for observational data: load.data
- for each model run: read.output
- align the data
- for each metric: calculate the metric and record scores in results object

Draft flowchart:
(Feel free to make comments at https://www.lucidchart.com/invitations/accept/218cb9e0-cc31-4c68-8ea8-c20adfc195d6) 

![calc benchmark - calc metrics](https://cloud.githubusercontent.com/assets/5503595/9537120/a5748b60-4d02-11e5-9703-4442ffce50b7.png)

**Comment by bcow** [comment]
@tonygardella can you add this to the benchmarking milestone?

**Comment by mdietze** [comment]
@robkooper Another thing to note on this issue -- in the PEcAn2 proposal we discussed the possibility of a REST API being used not just for getting whole or partial files, but also for moving calculations to where the data is. Betsy and I discussed this function as playing an initial role in that approach -- essentially limiting the set of possible remote computations that are allowed to those that are pre-existing Metric functions. At least as a starting point, this seems a lot simpler and safer than allowing external users to push arbitrary scripts to any server in the PEcAn network!
